File: app/src/main/java/com/worldventures/dreamtrips/wallet/analytics/settings/SettingsOfflineModeStateChangeAction.java
package com.worldventures.dreamtrips.wallet.analytics.settings;

import com.worldventures.core.service.analytics.ActionPart;
import com.worldventures.core.service.analytics.AdobeTracker;
import com.worldventures.core.service.analytics.AnalyticsEvent;
import com.worldventures.core.service.analytics.AttributeMap;
import com.worldventures.dreamtrips.wallet.analytics.WalletAnalyticsAction;

import java.util.HashMap;
import java.util.Map;

@AnalyticsEvent(action = "wallet:settings:security:offline mode:${offlineModeState}",
                trackers = AdobeTracker.TRACKER_KEY)
public class SettingsOfflineModeStateChangeAction extends WalletAnalyticsAction {

   @ActionPart String offlineModeState;

   @AttributeMap final Map<String, String> attributeMap = new HashMap<>();

   public SettingsOfflineModeStateChangeAction(boolean offlineModeState) {
      this.offlineModeState = offlineModeState ? "enable" : "disable";
      attributeMap.put(offlineModeState ? "offlinemodeenabled" : "offlinemodedisabled", String.valueOf(1));
   }

}

File: wallet/src/main/java/com/worldventures/wallet/analytics/settings/SettingsOfflineModeStateChangeAction.kt
package com.worldventures.wallet.analytics.settings

import com.worldventures.core.service.analytics.ActionPart
import com.worldventures.janet.analytics.AnalyticsEvent
import com.worldventures.core.service.analytics.AttributeMap
import com.worldventures.core.service.analytics.AdobeTracker
import com.worldventures.wallet.analytics.WalletAnalyticsAction

import java.util.HashMap

@AnalyticsEvent(action = "wallet:settings:security:offline mode:\${offlineModeState}", trackers = arrayOf(AdobeTracker.TRACKER_KEY))
class SettingsOfflineModeStateChangeAction(offlineModeState: Boolean) : WalletAnalyticsAction() {

   @ActionPart internal var offlineModeState: String = if (offlineModeState) "enable" else "disable"

   @AttributeMap internal val attributeMap: MutableMap<String, String> = HashMap()

   init {
      attributeMap.put(if (offlineModeState) "offlinemodeenabled" else "offlinemodedisabled", 1.toString())
   }

}
